prijateli_tree/app/main.py

```diff
@@ -43,3 +44,9 @@
 @app.get("/", response_class=HTMLResponse)
 def home(request: Request) -> Response:
     return templates.TemplateResponse("home_page.html", {"request": request})
+
+
+@app.on_event("shutdown")
+def close_external_connections():
+    # Close database connection on application shutdown
+    Database.close_connection()
```

Comment:
```commandline
2024-01-05 19:06:43 prijatelitree-web-1       | INFO:     192.168.65.1:46073 - "GET / HTTP/1.1" 200 OK
2024-01-05 19:06:43 prijatelitree-web-1       | INFO:     192.168.65.1:46073 - "GET /css/bootstrap.min.css HTTP/1.1" 304 Not Modified
2024-01-05 19:06:43 prijatelitree-web-1       | INFO:     192.168.65.1:46075 - "GET /css/custom.css HTTP/1.1" 304 Not Modified
2024-01-05 19:06:43 prijatelitree-web-1       | INFO:     192.168.65.1:46074 - "GET /js/bootstrap.bundle.min.js HTTP/1.1" 304 Not Modified
2024-01-05 19:06:43 prijatelitree-web-1       | INFO:     192.168.65.1:46077 - "GET /js/jquery.dataTables.min.js HTTP/1.1" 304 Not Modified
2024-01-05 19:06:43 prijatelitree-web-1       | INFO:     192.168.65.1:46073 - "GET /images/logo.png HTTP/1.1" 304 Not Modified
2024-01-05 19:06:43 prijatelitree-web-1       | INFO:     192.168.65.1:46078 - "GET /js/dataTables.bootstrap5.min.js HTTP/1.1" 304 Not Modified
2024-01-05 19:06:43 prijatelitree-web-1       | INFO:     192.168.65.1:46076 - "GET /js/jquery-3.6.0.min.js HTTP/1.1" 304 Not Modified
2024-01-05 19:06:51 prijatelitree-web-1       | INFO:     Shutting down
2024-01-05 19:06:51 prijatelitree-web-1       | INFO:     Waiting for application shutdown.
2024-01-05 19:06:51 prijatelitree-web-1       | INFO:   Database connection closed at 2024-01-06 01:06:51.937285
2024-01-05 19:06:51 prijatelitree-web-1       | INFO:     Application shutdown complete.
2024-01-05 19:06:51 prijatelitree-web-1       | INFO:     Finished server process [8]
2024-01-05 19:06:52 prijatelitree-web-1       | INFO:     Stopping reloader process [1]
prijatelitree-web-1 exited with code 0
prijatelitree-web-1 exited with code 0
prijatelitree-postgres-1  | 2024-01-06T01:06:52.156633509Z 2024-01-06 01:06:52.156 UTC [1] LOG:  received fast shutdown request
2024-01-05 19:06:52 prijatelitree-postgres-1  | 2024-01-06 01:06:52.158 UTC [1] LOG:  aborting any active transactions
2024-01-05 19:06:52 prijatelitree-postgres-1  | 2024-01-06 01:06:52.159 UTC [1] LOG:  background worker "logical replication launcher" (PID 32) exited with exit code 1
2024-01-05 19:06:52 prijatelitree-postgres-1  | 2024-01-06 01:06:52.160 UTC [27] LOG:  shutting down
2024-01-05 19:06:52 prijatelitree-postgres-1  | 2024-01-06 01:06:52.160 UTC [27] LOG:  checkpoint starting: shutdown immediate
2024-01-05 19:06:52 prijatelitree-postgres-1  | 2024-01-06 01:06:52.163 UTC [27] LOG:  checkpoint complete: wrote 3 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.001 s, sync=0.001 s, total=0.004 s; sync files=2, longest=0.001 s, average=0.001 s; distance=0 kB, estimate=0 kB; lsn=0/1A027C0, redo lsn=0/1A027C0
2024-01-05 19:06:52 prijatelitree-postgres-1  | 2024-01-06 01:06:52.165 UTC [1] LOG:  database system is shut down
prijatelitree-postgres-1 exited with code 0
```

Reply:
When does a shutdown event occur?

Reply:
The shutdown event occurs when the FastAPI application closes. I simulated that using the `make stop` command. For more information you can look here: https://fastapi.tiangolo.com/advanced/events/#shutdown-event